CMakeLists.txt

```diff
@@ -657,6 +664,38 @@ include(cmake/OpenCVCompilerOptions.cmake)
 
 ocv_cmake_hook(POST_COMPILER_OPTIONS)
 
+# --- CUDA Support ---
+if(ENABLE_CUDA_FIRST_CLASS_LANGUAGE)
+  if(CMAKE_VERSION VERSION_LESS 3.18)
+    message(WARNING "CUDA first class language only supported for CMake versions >= 3.18, falling back to FindCUDA!")
+    set(ENABLE_CUDA_FIRST_CLASS_LANGUAGE OFF CACHE BOOL "Enable CUDA as a first class language, if enabled dependant projects will need to use CMake >= 3.18" FORCE)
+  else()
+    if(UNIX AND CUDA_PATH AND NOT ENV{CUDA_PATH})
+      set(ENV{CUDA_PATH} ${CUDA_PATH})
+    elseif(WIN32 AND CUDA_PATH)
+      set(ENV{PATH} "${CUDA_PATH}\\bin\;$ENV{PATH}")
```

Comment:
I cannot get this WIN32 path to work if CUDA_PATH is not set as an environment variable, setting CUDA_PATH as cmake variable does not work, check_language(CUDA) does not find the compiler. 

If [this](https://github.com/Kitware/CMake/blob/69f0dd42b063296ab96a3ab61392b36cadab04b8/Modules/CMakeDetermineCUDACompiler.cmake#L21) is what is run by check_language(CUDA) it only seems to look for the nvcc compiler in $ENV{CUDACXX} and $ENV{CUDA_PATH}/bin.

Should ENV{CUDA_PATH} be set for windows too as done for UNIX two lines above?

Reply:
> I cannot get this WIN32 path to work if CUDA_PATH is not set as an environment variable, setting CUDA_PATH as cmake variable does not work, check_language(CUDA) does not find the compiler.

This appears to be Visual Studio specific.  If I use the Ninja generator then instead then this works.

VS appears to (I'm still checkig this now) rely on a hard coded environmental variable, which if deleted prevents it from finding the CUDA compiler. In your case this would be **CUDA_PATH_V12_3**.

Reply:
Wow thank you, you are right, I didn't realize I was using Visual Studio CMake generator, conan selects it by default.
My environment variable CUDA_PATH were set correctly and nvcc was in PATH but nothing worked.
Switching to Ninja worked!

Reply:
If you want to use the VS generator then you can pass the location of your CUDA toolkit directly to CMake using the [toolset specification](https://cmake.org/cmake/help/latest/variable/CMAKE_GENERATOR_TOOLSET.html#visual-studio-toolset-selection) (see [this](https://discourse.cmake.org/t/cuda-selction-with-visual-studio-generator/9535/3) post for more details). e.g.
```
-Tcuda="C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.3"
```